specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx

```diff
@@ -197,12 +204,15 @@ export function QueryResults(props: Props): JSX.Element {
           </Button.Small>
         )}
         <div className="-ml-2 flex-1" />
-        {extraButtons}
+        {props.displayedFields.length > 0 && visibleFieldSpecs.length > 0
```

Comment:
This PR appears to be a superset of https://github.com/specify/specify7/pull/3848/files, and to that end the same comment applies

Reply:
what is the difference between `props.displayedFields.length` and `visibleFieldSpecs.length`? can you check if they are ever not equal, and if so, in which cases?

then, double check all other usages of `visibleFieldSpecs.length` and `props.displayedFields.length` in this file. Ideally, both lengths are always the same, and so you would only use one of those two in all places in this component

Reply:
> what is the difference between props.displayedFields.length and visibleFieldSpecs.length? can you check if they are ever not equal, and if so, in which cases?


https://github.com/specify/specify7/blob/2a259ca22b439ba71b9313705cb2a253ac8edd6e/specifyweb/frontend/js_src/lib/components/QueryBuilder/ResultsWrapper.tsx#L156

displayedFields seems to be the fields which are supposed to be shown in the Query Results (the ones that are explicitly marked to be shown in the results. This definition seems consistent for all usages of `displayedFields`)

https://github.com/specify/specify7/blob/2a259ca22b439ba71b9313705cb2a253ac8edd6e/specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx#L96

visibleFieldSpecs are all of the fieldSpecs that are not 'Phantom' fields. 

https://github.com/specify/specify7/blob/2a259ca22b439ba71b9313705cb2a253ac8edd6e/specifyweb/frontend/js_src/lib/components/QueryBuilder/fieldSpec.ts#L75-L82